CMakeLists.txt

```diff
@@ -60,6 +61,9 @@ cmake_dependent_option(AF_WITH_IMAGEIO "Build ArrayFire with Image IO support" $
                        "FreeImage_FOUND" OFF)
 cmake_dependent_option(AF_BUILD_FRAMEWORK "Build an ArrayFire framework for Apple platforms.(Experimental)" OFF
                        "APPLE" OFF)
+cmake_dependent_option(AF_WITH_LOGGING "Build ArrayFire with support for logging functionallity." ON
+                       "TARGET spdlog::spdlog" OFF)
+
```

Comment:
Based on the PR description, I expected `AF_WITH_LOGGING` to be an normal option instead of dependent option. If the target build machine has spdlog installed but wasn't intended for ArrayFire, we would still build ArrayFire with logging enabled. Is that the intended behavior we want to provide ?

Reply:
Yeah, good point. Will fix